app/src/main/java/com/asfoundation/wallet/subscriptions/SubscriptionDetailsFragment.kt

```diff
@@ -93,8 +94,27 @@ class SubscriptionDetailsFragment : DaggerFragment(), SubscriptionDetailsView {
     total_value_appc.text = String.format("~%s / APPC",
         subscriptionDetails.appcValue.setScale(FIAT_SCALE, RoundingMode.FLOOR))
 
-    next_payment_value.text = getDateString(subscriptionDetails.nextPayment)
     layout_active_subscription_content.payment_method_value.text = subscriptionDetails.paymentMethod
+
+
+    if (subscriptionDetails.expiresOn != null) {
+      expires_on.visibility = View.VISIBLE
+      cancel_subscription.visibility = View.GONE
+      next_payment_value.text = "Canceled"
+      next_payment_value.setTextColor(resources.getColor(R.color.red))
+
+      val dateFormat = SimpleDateFormat("MMM yy", Locale.getDefault())
+
+      expires_on.text =
+          String.format("Canceled - It expires on %s",
+              dateFormat.format(subscriptionDetails.expiresOn))
+
+      info.visibility = View.GONE
+      info_text.visibility = View.GONE
```

Comment:
Extract this to a method

Reply:
[MOB-46] Updated screens after review feedback